lib/rules/jsx-curly-brace-presence.js

```diff
@@ -62,6 +62,7 @@ module.exports = {
   },
 
   create(context) {
+    const HTML_ENTITY_REGEX = /&[A-Za-z\d#]+;/g;
```

Comment:
global regexes are stateful when used with `test`; we should make this a function that always returns a new one, if it's global.